### DISCUSSION TITLE
Build issues

### DISCUSSION THREAD

**Discussion by javedfaruquee1** [apache/camel-karavan#808] [Q&A]
Getting below error while trying to build locally with commands "mvn clean package -Dquarkus.container-image.build=true" provided in DEV.MD
Any suggestions here??

[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.16.6.Final:build (default) on project karavan: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkiverse.quinoa.deployment.QuinoaProcessor#processBuild threw an exception: java.lang.RuntimeException: Error in Quinoa while running package manager build command: npm.cmd run build
[ERROR]         at io.quarkiverse.quinoa.deployment.packagemanager.PackageManager.build(PackageManager.java:56)
[ERROR]         at io.quarkiverse.quinoa.deployment.QuinoaProcessor.processBuild(QuinoaProcessor.java:138)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:829)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)

**Comment by mgubaidullin** [comment]
DEV.md is quite outdated. I would recommend to take a look on Github Actions https://github.com/apache/camel-karavan/blob/main/.github/workflows/app.yml

**Reply by javedfaruquee1** [reply]
There are still issue , for example npm ci results in tests failures which were avoided after removing npm run test from package.json also mvn package -f karavan-app -Dquarkus.profile=oidc -DskipTest -Dquarkus.quinoa.package-manager-install=false  -Dquarkus.container-image.build=true  to build the app is failing


[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.16.7.Final:build (default) on project karavan: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkiverse.quinoa.deployment.QuinoaProcessor#prepareQuinoaDirectory threw an exception: java.util.ServiceConfigurationError: jakarta.json.spi.JsonProvider: org.glassfish.json.JsonProviderImpl not a subtype
[ERROR]         at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:589)
[ERROR]         at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1237)
[ERROR]         at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1265)
[ERROR]         at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1300)
[ERROR]         at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1385)
[ERROR]         at jakarta.json.spi.JsonProvider.provider(JsonProvider.java:124)
[ERROR]         at jakarta.json.Json.createReader(Json.java:204)
[ERROR]         at io.quarkiverse.quinoa.deployment.packagemanager.FrameworkType.detectFramework(FrameworkType.java:82)
[ERROR]         at io.quarkiverse.quinoa.deployment.QuinoaProcessor.prepareQuinoaDirectory(QuinoaProcessor.java:110)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:829)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)

**Comment by mgubaidullin** [comment]
Have you followed the github action?
1. Generate Karavan UI model and API, Kamelets and Components
2. Karavan Core install
3. Build application

**Reply by spyhunter99** [reply]
was the pom in `karavan-app` deleted? i've got nothing there

**Comment by mgubaidullin** [comment]
Latest stable branch is 3.21.0 https://github.com/apache/camel-karavan/tree/3.21.0
main branch is in active development and refactoring

**Comment by davsclaus** [comment]
closing old tickets
